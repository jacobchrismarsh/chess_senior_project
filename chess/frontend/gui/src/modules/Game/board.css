.game-board {
  outline-style: solid;
}

.tri-list {
  display: flex;
  flex-direction: column;
  align-content: space-between;
  justify-content: space-between;
}

.list-element {
}
