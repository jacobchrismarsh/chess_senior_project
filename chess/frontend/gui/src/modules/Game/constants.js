export const NOT_SELECTED = -1;

// Colors
export const WHITE = "White";
export const BLACK = "Black";

// Init Piece Positions
export const BLACK_PAWN_INIT = [8, 9, 10, 11, 12, 13, 14, 15];
export const WHITE_PAWN_INIT = [48, 49, 50, 51, 52, 53, 54, 55];
export const BLACK_KNIGHT_INIT = [1, 6];
export const WHITE_KNIGHT_INIT = [57, 62];
export const BLACK_ROOK_INIT = [0, 7];
export const WHITE_ROOK_INIT = [56, 63];
export const BLACK_BISHOP_INIT = [2, 5];
export const WHITE_BISHOP_INIT = [58, 61];
export const BLACK_QUEEN_INIT = [3];
export const WHITE_QUEEN_INIT = [59];
export const BLACK_KING_INIT = [4];
export const WHITE_KING_INIT = [60];

export const TRANSLATE_POSITION = {
  0: 56,
  1: 57,
  2: 58,
  3: 59,
  4: 60,
  5: 61,
  6: 62,
  7: 63,
  8: 48,
  9: 49,
  10: 50,
  11: 51,
  12: 52,
  13: 53,
  14: 54,
  15: 55,
  16: 40,
  17: 41,
  18: 42,
  19: 43,
  20: 44,
  21: 45,
  22: 46,
  23: 47,
  24: 32,
  25: 33,
  26: 34,
  27: 35,
  28: 36,
  29: 37,
  30: 38,
  31: 39,
  32: 24,
  33: 25,
  34: 26,
  35: 27,
  36: 28,
  37: 29,
  38: 30,
  39: 31,
  40: 16,
  41: 17,
  42: 18,
  43: 19,
  44: 20,
  45: 21,
  46: 22,
  47: 23,
  48: 8,
  49: 9,
  50: 10,
  51: 11,
  52: 12,
  53: 13,
  54: 14,
  55: 15,
  56: 0,
  57: 1,
  58: 2,
  59: 3,
  60: 4,
  61: 5,
  62: 6,
  63: 7
}
