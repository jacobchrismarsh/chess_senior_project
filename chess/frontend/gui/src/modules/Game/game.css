ul {
  list-style-type: none;
  resize: none;
}

li {
  display: inline;
  resize: none;
}

.game-container {
  height: 100vh !important;
  display: flex;
  flex-direction: row;
  align-items: center;
}

.spacer {
  height: 44%;
  margin: 25px;
  flex-grow: 1;
  border-radius: 5px;
}

.sidebar {
  font-family: "Courier New", Courier, monospace;
  height: 44%;
  width: 25%;
  margin: 25px;
  flex-grow: 1;
  opacity: 0.8;
  background-color: antiquewhite;
  border-radius: 5px;
}

.game-information {
  display: flex;
  align-items: center;
  flex-direction: column;
}

.left-sidebar-container {
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;
}

.left-sidebar-item {
  height: 50%;
}

.color-black {
  background-color: black;
}
