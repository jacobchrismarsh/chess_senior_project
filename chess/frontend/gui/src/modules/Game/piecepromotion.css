.overlay {
  width: 25%;
  height: 25%;
  top: 10%; 
  z-index: 2; /* Specify a stack order in case you're using a different order for other elements */
  position: absolute;
  text-align: center;
  top: 40%;
  left: 37.5%;
  background-color: antiquewhite;
  border-radius: 5px;
}

.overlay-container {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  margin-top: 10%;
}