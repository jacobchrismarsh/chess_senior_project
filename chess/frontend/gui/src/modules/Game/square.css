.square {
  background: transparent;
  border: 2px;
  border-color: RGB(0, 0, 0);
  float: left;
  height: 50px;
  margin-right: -1px;
  margin-top: -1px;
  padding: 0;
  text-align: center;
  width: 50px;
}

.selected {
  background-color: beige !important;
}

.movable {
  background-color: lightgreen !important;
}

.square:focus {
  outline: none;
}

.light {
  background-color: RGB(255, 255, 255);
}

.dark {
  background-color: RGB(75, 75, 75);
}
