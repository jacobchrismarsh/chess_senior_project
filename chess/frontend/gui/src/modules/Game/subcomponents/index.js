export * from "./Board";
export * from "./Square";
export * from "./RightSidebar";
export * from "./LeftSidebar";
export * from "./Clock";
export * from "./PiecePromotion";
