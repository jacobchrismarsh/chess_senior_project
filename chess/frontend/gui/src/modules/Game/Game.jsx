import React from "react";
import autoBind from 'react-autobind';
import { Board, LeftSidebar, RightSidebar, PiecePromotion } from "./subcomponents";
import {
  Pawn,
  Knight,
  Rook,
  Bishop,
  Queen,
  King,
  Empty
} from "./subcomponents/Pieces";
import {
  NOT_SELECTED,
  BLACK_PAWN_INIT,
  WHITE_PAWN_INIT,
  BLACK_KNIGHT_INIT,
  WHITE_KNIGHT_INIT,
  BLACK_ROOK_INIT,
  WHITE_ROOK_INIT,
  BLACK_BISHOP_INIT,
  WHITE_BISHOP_INIT,
  BLACK_QUEEN_INIT,
  WHITE_QUEEN_INIT,
  BLACK_KING_INIT,
  WHITE_KING_INIT,
  WHITE,
  BLACK,
  TRANSLATE_POSITION,
  ONE_MINUTE
} from "./constants";
import $ from "jquery";
import "./game.css";

export default class Game extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      squares: this.initBoard(),
      selected: NOT_SELECTED,
      capturedWhitePieces: [],
      capturedBlackPieces: [],
      turn: WHITE,
      count: 0,
      error: "",
      movableSquares: [],
      whiteTime: (Date.now() + (ONE_MINUTE * 10)),
      blackTime: (Date.now() + (ONE_MINUTE * 10)),
      showPiecePromotion: false,
      promotedPiece: null,
      yourColor: WHITE
    };

    autoBind(this);
  }

  initBoard() {
    // Not using a 2D array just for simplicity
    let squares = Array(64).fill(null);

    // // Initiailizing the 'Black' Pawns
    for (let i = 0; i < BLACK_PAWN_INIT.length; i++) {
      squares[BLACK_PAWN_INIT[i]] = new Pawn(BLACK);
    }

    // Initializing the 'White' Pawns
    for (let i = 0; i < WHITE_PAWN_INIT.length; i++) {
      squares[WHITE_PAWN_INIT[i]] = new Pawn(WHITE);
    }

    // Initializing the 'Black' Knights
    for (let i = 0; i < BLACK_KNIGHT_INIT.length; i++) {
      squares[BLACK_KNIGHT_INIT[i]] = new Knight(BLACK);
    }

    // Initializing the 'White' Knights
    for (let i = 0; i < WHITE_KNIGHT_INIT.length; i++) {
      squares[WHITE_KNIGHT_INIT[i]] = new Knight(WHITE);
    }

    // Initializing the 'Black' Rooks
    for (let i = 0; i < BLACK_ROOK_INIT.length; i++) {
      squares[BLACK_ROOK_INIT[i]] = new Rook(BLACK);
    }

    // Initialzigin the 'White' Rooks
    for (let i = 0; i < WHITE_ROOK_INIT.length; i++) {
      squares[WHITE_ROOK_INIT[i]] = new Rook(WHITE);
    }

    // Initializing the 'Black' Bishops
    for (let i = 0; i < BLACK_BISHOP_INIT.length; i++) {
      squares[BLACK_BISHOP_INIT[i]] = new Bishop(BLACK);
    }

    // Initializing the 'White' Bishops
    for (let i = 0; i < WHITE_BISHOP_INIT.length; i++) {
      squares[WHITE_BISHOP_INIT[i]] = new Bishop(WHITE);
    }

    // Initializing the 'Black' Queen
    for (let i = 0; i < BLACK_QUEEN_INIT.length; i++) {
      squares[BLACK_QUEEN_INIT[i]] = new Queen(BLACK);
    }
    // Initializing the 'White' Queen
    for (let i = 0; i < WHITE_QUEEN_INIT.length; i++) {
      squares[WHITE_QUEEN_INIT[i]] = new Queen(WHITE);
    }

    // Initialzing the 'Black' King
    for (let i = 0; i < BLACK_KING_INIT.length; i++) {
      squares[BLACK_KING_INIT[i]] = new King(BLACK);
    }

    // Initializing the 'White' King
    for (let i = 0; i < WHITE_KING_INIT.length; i++) {
      squares[WHITE_KING_INIT[i]] = new King(WHITE);
    }

    for (let i = 0; i < 64; i++) {
      if (squares[i] === null) {
        squares[i] = new Empty(null);
      }
    }

    return squares;
  }

  sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
  }

  componentDidMount() {
    // if your color is black then you wait for the opponent to go first
    if (this.state.yourColor === BLACK) {
      this.sleep(1000).then(() => {
          this.getOpponentMove();
        }
      );
    }
  }

  // returns a boolean indication whether a move is valid
  checkIfMoveValid(index) {
    return this.state.squares[index].movable;
  }

  // returns the color of the opponent
  opponentColor() {
    return this.state.turn === WHITE ? BLACK : WHITE;
  }

  // sets the error message to the string that is passed in
  setError(message) {
    this.setState({
      error: message
    });
  }

  // returns true if the square at spot index belongs to the current player
  checkIfPieceBelongsToCurrentPlayer(index) {
    return this.state.squares[index].player === this.state.turn;
  }

  // returns true if the square at the spot index belongs to the opponent player
  checkIfPieceBelongsToOpponentPlayer(index) {
    return this.state.squares[index].player === this.opponentColor();
  }

  // returns true if the square at spot index is already selected
  checkIfSelectedPieceIsAlreadySelected(index) {
    return this.state.selected === index;
  }

  // returns true if the square at spot index is empty
  checkIfSelectedIsEmpty(index) {
    return this.state.squares[index].player === null;
  }

  // highlights the squares on the board that are 'movable' according
  // to the response from the server
  highlightMoves(moves) {
    let newSquares = this.state.squares;

    moves.forEach(move => {
      newSquares[TRANSLATE_POSITION[move]].canMovePiece();
    });

    this.setState({
      squares: newSquares
    });
  }

  // unhighlights the 'movable' pieces on the board
  dehighlightMoves() {
    let newStateSquares = this.state.squares;
    newStateSquares.forEach(square => {
      if (square.movable) {
        square.cannotMovePiece();
      }
    });

    this.setState({
      squares: newStateSquares
    });
  }

  checkPossibleMoves(index) {
    return $.ajax({
      url: "http://127.0.0.1:8000/game/get_moves/",
      method: "GET",
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `JWT ${localStorage.getItem('token')}`
      },
      data: { index: TRANSLATE_POSITION[index] }
    }).then(response => {
      this.highlightMoves(response.moves);
    });
  }

  checkIfCaptureWhite(turn, opponentColor, squares, index) {
    return turn === WHITE && opponentColor === squares[TRANSLATE_POSITION[index]].player
  }

  checkIfCaptureBlack(turn, opponentColor, squares, index) {
    return turn === BLACK && opponentColor === squares[TRANSLATE_POSITION[index]].player
  }

  // function used to update the board with the moves from the 
  // backend's response
  updateBoardWithMove(response) {
    let newStateSquares = this.state.squares;
    let capturedB = this.state.capturedBlackPieces;
    let capturedW = this.state.capturedWhitePieces;
    let turn = this.state.turn
    let opponentColor = this.opponentColor();
    response.moves.forEach(move => {
      let from_coord_response = move.from_coord
      let to_coord_response = move.to_coord

      // if it is white's turn and you are placing on top of an opponent's piece we need to update
      // the capture array to show we captured a black piece
      if (this.checkIfCaptureWhite(turn, opponentColor, newStateSquares, to_coord_response)) {
        capturedB.push(newStateSquares[TRANSLATE_POSITION[to_coord_response]]);
      }

      // if it is black's turn and you are placing on top of an opponent's piece we need to update
      // the capture array to show we captured a white piece
      if (this.checkIfCaptureBlack(turn, opponentColor, newStateSquares, to_coord_response)) {
        capturedW.push(newStateSquares[TRANSLATE_POSITION[to_coord_response]]);
      }

      let pieceToMove = this.state.squares[TRANSLATE_POSITION[from_coord_response]];
      newStateSquares[TRANSLATE_POSITION[to_coord_response]] = pieceToMove;
      newStateSquares[TRANSLATE_POSITION[from_coord_response]] = new Empty(null);
    });
    
    this.setState({
      squares: newStateSquares,
      turn: this.opponentColor(),
      capturedBlackPieces: capturedB,
      capturedWhitePieces: capturedW
    });
  }

  makeMove(from_coord, to_coord) {
    return $.ajax({
      url: "http://127.0.0.1:8000/game/make_move/",
      method: "GET",
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `JWT ${localStorage.getItem('token')}`
      },
      data: { 
        from_coord: TRANSLATE_POSITION[from_coord],
        to_coord: TRANSLATE_POSITION[to_coord]
      }
    }).then(response => {
      this.updateBoardWithMove(response)
    });
  }

  getOpponentMove() {
    return $.ajax({
      url: "http://127.0.0.1:8000/game/get_opponent_move/",
      method: "GET",
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `JWT ${localStorage.getItem('token')}`
      },
      data: {
        player_color: this.state.turn
      }
    }).then(response => {
      this.updateBoardWithMove(response)
    });
  }

  // Function used to handle the piece moving, rn the
  // capture boolean doesn't do anything but I will use
  // it to keep track of pieces that have been captured
  handleMove(capture, index) {
    let preMovePosition = this.state.squares;
    let pieceToMove = this.state.squares[this.state.selected];
    let postMovePosition = preMovePosition;
    let capturedB = this.state.capturedBlackPieces;
    let capturedW = this.state.capturedWhitePieces;
    // If the piece was captured and it is White's turn then the
    // captured piece is black
    if (capture && this.state.turn === WHITE) {
      capturedB.push(preMovePosition[index]);
    }
    // If the piece was captured and it is Black's turn then the
    // captured piece is white
    else if (capture && this.state.turn === BLACK) {
      capturedW.push(preMovePosition[index]);
    }

    if (!this.checkIfMoveValid(index)) {
      this.setError("Cannot move piece there");
    } else {
      pieceToMove.deselectPiece();
      this.dehighlightMoves();
      this.makeMove(this.state.selected, index);
      
      postMovePosition[index] = pieceToMove;
      
      this.setState({
        squares: postMovePosition,
        selected: NOT_SELECTED,
        capturedBlackPieces: capturedB,
        capturedWhitePieces: capturedW,
        count: this.state.count + 1,
        error: ""
      });

      // Finish off by pinging for the next move that I am going to recieve
      this.getOpponentMove();
    }
  }

  // Function that is passed down to the child components used
  // to handle moving the pieces
  handleClick(index) {
    let newStateSquares = this.state.squares;
    // if the player currently has no piece selected
    if (this.state.selected === NOT_SELECTED) {
      // if the spot that was clicked was the current player's color
      if (this.checkIfPieceBelongsToCurrentPlayer(index)) {
        this.checkPossibleMoves(index);
        newStateSquares[index].selectPiece();
        this.setState({
          squares: newStateSquares,
          selected: index,
          error: ""
        });
      } else if (this.checkIfPieceBelongsToOpponentPlayer(index)) {
        // if the spot that was clicked was the opponent's color
        this.setError("Cannot move opponent's piece");
      }
    } else {
      // if the player currently has a piece selected
      // check the spot that was clicked

      // if the spot that was clicked was the current player's color
      if (this.checkIfPieceBelongsToCurrentPlayer(index)) {
        // if the spot selected is the current selected piece, then we just need to
        // deselect the piece
        if (this.checkIfSelectedPieceIsAlreadySelected(index)) {
          this.dehighlightMoves();
          newStateSquares[index].deselectPiece();
          this.setState({
            squares: newStateSquares,
            selected: NOT_SELECTED,
            error: ""
          });
        } else {
          // if the spot selected is not the current selected piece throw an error,
          // since you cannot capture your own piece
          this.setError("Cannot cannot capture your own piece");
        }
      } else if (this.checkIfPieceBelongsToOpponentPlayer(index)) {
        // if the spot that was clcked was the opposite player's color
        let captured = true;
        this.handleMove(captured, index);
      } else if (this.checkIfSelectedIsEmpty(index)) {
        // if the spot that was clicked was an empty spot
        let captured = false;
        this.handleMove(captured, index);
      }
    }
  }

  handlePiecePromotion(index) {
    let pieces = ['Queen', 'Rook', 'Knight', 'Bishop']

    this.setState({
      promotedPiece: pieces[index],
      showPiecePromotion: false
    });
  }

  render() {
    return (
      <div className="game-container">
        <div className="spacer" />
        
        <PiecePromotion
          show={this.state.showPiecePromotion}
          color={this.state.turn}
          handleClick={this.handlePiecePromotion}
        />

        <LeftSidebar
          white={this.state.capturedWhitePieces}
          black={this.state.capturedBlackPieces}
        />
        <Board
          yourColor={this.state.yourColor}
          squares={this.state.squares}
          handleClick={this.handleClick}
          turn={this.state.turn}
          move={this.state.count}
          whiteTime={this.state.whiteTime}
          blackTime={this.state.blackTime}
        />
        <RightSidebar
          turn={this.state.turn}
          count={this.state.count}
          error={this.state.error}
        />
        <div className="spacer" />
      </div>
    );
  }
}
