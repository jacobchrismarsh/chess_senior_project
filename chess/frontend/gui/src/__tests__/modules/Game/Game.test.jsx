import React from 'react';
import { shallow } from 'enzyme';
import Game from "../../../modules/Game/Game";

describe("modules/Game/Game.jsx", () => {
  let game = null;

  beforeEach(() => {
    const realDateNow = Date.now.bind(global.Date);
    const dateNowStub = jest.fn(() => 1530518207007);
    global.Date.now = dateNowStub;
    
    game = shallow(<Game />);
  });

  it("should give an initial board", () => {
    let squares = game.instance().initBoard();
    expect(squares).toEqual(
      [
        {"movable": false, "player": "Black", "selected": false, "svg": "br.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bn.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bb.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bq.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bk.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bb.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bn.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "br.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bp.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bp.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bp.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bp.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bp.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bp.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bp.svg"},
        {"movable": false, "player": "Black", "selected": false, "svg": "bp.svg"},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": null, "selected": false, "svg": undefined},
        {"movable": false, "player": "White", "selected": false, "svg": "wp.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wp.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wp.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wp.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wp.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wp.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wp.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wp.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wr.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wn.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wb.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wq.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wk.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wb.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wn.svg"},
        {"movable": false, "player": "White", "selected": false, "svg": "wr.svg"}]);
  });

  it("should render the Game component", () => {
    expect(game).toMatchSnapshot();
  });
});