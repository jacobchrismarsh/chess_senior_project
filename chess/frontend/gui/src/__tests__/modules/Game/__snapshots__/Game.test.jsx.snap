// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/Game/Game.jsx should render the Game component 1`] = `
<div
  className="game-container"
>
  <div
    className="spacer"
  />
  <PiecePromotion
    color="White"
    handleClick={[Function]}
    show={false}
  />
  <GameOver
    show={false}
    winner={null}
  />
  <LeftSidebar
    black={Array []}
    white={Array []}
  />
  <Board
    blackTime={1530518807007}
    handleClick={[Function]}
    move={0}
    squares={
      Array [
        Rook {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "br.svg",
        },
        Knight {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bn.svg",
        },
        Bishop {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bb.svg",
        },
        Queen {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bq.svg",
        },
        King {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bk.svg",
        },
        Bishop {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bb.svg",
        },
        Knight {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bn.svg",
        },
        Rook {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "br.svg",
        },
        Pawn {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bp.svg",
        },
        Pawn {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bp.svg",
        },
        Pawn {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bp.svg",
        },
        Pawn {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bp.svg",
        },
        Pawn {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bp.svg",
        },
        Pawn {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bp.svg",
        },
        Pawn {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bp.svg",
        },
        Pawn {
          "movable": false,
          "player": "Black",
          "selected": false,
          "svg": "bp.svg",
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Empty {
          "movable": false,
          "player": null,
          "selected": false,
          "svg": undefined,
        },
        Pawn {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wp.svg",
        },
        Pawn {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wp.svg",
        },
        Pawn {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wp.svg",
        },
        Pawn {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wp.svg",
        },
        Pawn {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wp.svg",
        },
        Pawn {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wp.svg",
        },
        Pawn {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wp.svg",
        },
        Pawn {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wp.svg",
        },
        Rook {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wr.svg",
        },
        Knight {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wn.svg",
        },
        Bishop {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wb.svg",
        },
        Queen {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wq.svg",
        },
        King {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wk.svg",
        },
        Bishop {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wb.svg",
        },
        Knight {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wn.svg",
        },
        Rook {
          "movable": false,
          "player": "White",
          "selected": false,
          "svg": "wr.svg",
        },
      ]
    }
    turn="White"
    whiteTime={1530518807007}
    yourColor="White"
  />
  <RightSidebar
    count={0}
    error=""
    turn="White"
  />
  <div
    className="spacer"
  />
</div>
`;
